zebrad/tests/acceptance.rs

```diff
@@ -1052,3 +1055,170 @@ async fn tracing_endpoint() -> Result<()> {
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let listen_addr = format!("127.0.0.1:{}", port);
+
+    // Write a configuration that has our created metrics endpoint_addr
+    let mut config = default_test_config()?;
+    config.metrics.endpoint_addr = Some(listen_addr.parse().unwrap());
+    let dir1 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+    let regex1 = format!(r"Opened metrics endpoint at {}", listen_addr);
+
+    // From another folder create a configuration with the same endpoint.
+    // `metrics.endpoint_addr` will be the same in the 2 nodes.
+    // But they will have different Zcash listeners (auto port) and states (ephemeral)
+    let dir2 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, PORT_IN_USE_ERROR)?;
+
+    Ok(())
+}
+
+/// Start 2 zebrad nodes using the same tracing listener port, but different
+/// state directories and Zcash listener ports. The first node should get
+/// exclusive use of the port. The second node will panic with the Zcash tracing
+/// conflict hint added in #1535.
+#[test]
+fn zcash_tracing_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let listen_addr = format!("127.0.0.1:{}", port);
+
+    // Write a configuration that has our created tracing endpoint_addr
+    let mut config = default_test_config()?;
+    config.tracing.endpoint_addr = Some(listen_addr.parse().unwrap());
+    let dir1 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+    let regex1 = format!(r"Opened tracing endpoint at {}", listen_addr);
+
+    // From another folder create a configuration with the same endpoint.
+    // `tracing.endpoint_addr` will be the same in the 2 nodes.
+    // But they will have different Zcash listeners (auto port) and states (ephemeral)
+    let dir2 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, PORT_IN_USE_ERROR)?;
+
+    Ok(())
+}
+
+/// Start 2 zebrad nodes using the same state directory, but different Zcash
+/// listener ports. The first node should get exclusive access to the database.
+/// The second node will panic with the Zcash state conflict hint added in #1535.
+#[test]
+fn zcash_state_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // A persistent config has a fixed temp state directory, but asks the OS to
+    // automatically choose an unused port
+    let mut config = persistent_test_config()?;
+    let dir_conflict = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    let mut dir_conflict_full = PathBuf::new();
+    dir_conflict_full.push(dir_conflict.path());
+    dir_conflict_full.push("state");
+    dir_conflict_full.push("state");
+    dir_conflict_full.push("v4");
+    dir_conflict_full.push("mainnet");
```

Comment:
This test will break when we change the state version, or the network in the test config.

There's also an easier way to build the path using a slice:
https://doc.rust-lang.org/std/path/struct.PathBuf.html#examples

```suggestion
    let dir_conflict_full: PathBuf = [
        dir_conflict.path(),
        "state",
        "state",
        format!("v{}", zebra_state::DATABASE_FORMAT_VERSION),
        config.network.network.to_string().make_ascii_lowercase(),
    ].iter().collect();
```

Reply:
From the same documentation, the push is better for dynamic situation, specifically if we have variables of different types just as `Path` and `String`. Whatever is the case about that(which one is better) i added the version and the network data to the string in https://github.com/ZcashFoundation/zebra/pull/1535/commits/8f3c0f4663c74f2e8b562db1f438bf05a85c20a6

Reply:
Oh right yeah Rust doesn't really like `dyn` types in slices